### PR TITLE
Add scaffolding for packages page

### PR DESCRIPTION
**Pull request by isaac73**
# Type of change? 

- [ ] Bug fix
- [x] New feature
- [ ] Enhancement
- [ ] Breaking change (breaks backward compatibility)

# Description, Context, Motivation 
Add scaffolding for packages page
...

**Checklist:**

- [ ] All tests passed
- [ ] The code follows the code style and conventions of this project
- [ ] The change requires a change to the documentation
- [ ] I have updated the documentation accordingly

---
This change is [Reviewable](https://reviewable.io/reviews/azure/pcs-remote-monitoring-webui/1063)

### CODE REVIEW

File: src/components/pages/packages/packages.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+      t: this.props.t
+    };
+
+    return [
+      <ContextMenu key="context-menu">
+        {this.state.contextBtns}
+        <Protected permission={permissions.addPackage}>
+          <Btn svg={svgs.plus} onClick={this.appPackageFlyout}>{t('packages.flyouts.new.contextMenuName')}</Btn>
+        </Protected>
+        <ManageDeviceGroupsBtn />
```
[the remark this answers is not in the document]
Comment (isaac73) [reply src/components/pages/packages/packages.js]: This is according to spec. Device groups dropdown is not required here.

File: src/assets/icons/packages.svg [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<!-- Copyright (c) Microsoft. All rights reserved. -->
+<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 13.359 16">
+  <defs>
+    <style>
```
[the remark this answers is not in the document]
Comment (isaac73) [reply src/assets/icons/packages.svg]: done

File: src/assets/icons/packages.svg [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
...3,3.033,0,0,1-.086.737l1.055.437L13.4-12.71l-1.063-.437A3.1,3.1,0,0,1,11.3-12.11l.437,1.063-1.012.412-.437-1.055a3.033,3.033,0,0,1-.737.086,3.033,3.033,0,0,1-.737-.086l-.437,1.055L7.36-11.047,7.8-12.11a3.1,3.1,0,0,1-1.038-1.038L5.7-12.71l-.412-1.012,1.055-.437a3.033,3.033,0,0,1-.086-.737,3.033,3.033,0,0,1,.086-.737l-1.055-.437L5.7-17.083l1.063.437A3.1,3.1,0,0,1,7.8-17.684L7.36-18.747l1.012-.412L8.809-18.1a3.033,3.033,0,0,1,.737-.086,3.033,3.033,0,0,1,.737.086l.437-1.055,1.012.412L11.3-17.684a3.1,3.1,0,0,1,1.038,1.038l1.063-.437.412,1.012M9.546-12.7a2.1,2.1,0,0,0,.849-.176,2.248,2.248,0,0,0,.7-.472,2.262,2.262,0,0,0,.472-.695,2.091,2.091,0,0,0,.176-.853,2.1,2.1,0,0,0-.176-.849,2.248,2.248,0,0,0-.472-.7,2.248,2.248,0,0,0-.7-.472,2.1,2.1,0,0,0-.849-.176,2.091,2.091,0,0,0-.853.176A2.262,2.262,0,0,0,8-16.445a2.248,2.248,0,0,0-.472.7,2.1,2.1,0,0,0-.176.849,2.091,2.091,0,0,0,.176.853A2.262,2.262,0,0,0,8-13.349a2.262,2.262,0,0,0,.695.472,2.091,2.091,0,0,0,.853.176m6.68,3.841L14.253-6.9l-.635-.635.772-.78H2.961V-12.7h1.1v3.293H14.391l-.772-.78.635-.635L16.226-8.86M4.839-18.97,2.866-20.933,4.839-22.9l.635.635-.772.78h11.43v4.39h-1.1v-3.293H4.7l.772.78Z" transform="translate(-2.866 22.897)"/>
```
[the remark this answers is not in the document]
Comment (isaac73) [reply src/assets/icons/packages.svg]: done

File: src/components/pages/packages/flyouts/newPackage/newPackage.container.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+import { connect } from 'react-redux';
+import { translate } from 'react-i18next';
+import { NewPackage } from './newPackage';
+import { epics as packagesEpics } from 'store/reducers/packagesReducer';
+
+const mapStateToProps = state => ({});
```
[the remark this answers is not in the document]
Comment (isaac73) [reply src/components/pages/packages/flyouts/newPackage/newPackage.container.js]: done

File: src/components/app/app.js [hunk 5]
```diff
@@ -32,10 +33,11 @@ const dashboardTab     = { to: '/dashboard',     svg: svgs.tabs.dashboard,   lab
 const devicesTab       = { to: '/devices',       svg: svgs.tabs.devices,     labelId: 'tabs.devices' };
 const rulesTab         = { to: '/rules',         svg: svgs.tabs.rules,       labelId: 'tabs.rules' };
 const maintenanceTab   = { to: '/maintenance',   svg: svgs.tabs.maintenance, labelId: 'tabs.maintenance' };
+const packagesTab = {to: '/packages', svg: svgs.tabs.packages, labelId: 'tabs.packages'};
```
[the remark this answers is not in the document]
Comment (isaac73) [reply src/components/app/app.js]: why is this different from regular code standards? I'd prefer the default VS Code formatting.

File: src/components/pages/packages/flyouts/newPackage/newPackage.js [hunk 6]
```diff
@@ -0,0 +1,11 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+import {
```
[the remark this answers is not in the document]
Comment (isaac73) [reply src/components/pages/packages/flyouts/newPackage/newPackage.js]: done

File: src/components/pages/packages/flyouts/newPackage/newPackage.js [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+  LinkedComponent
+} from 'utilities';
+
+import './newPackage.css';
+
+export class NewPackage extends LinkedComponent {
+
```
[the remark this answers is not in the document]
Comment (isaac73) [reply src/components/pages/packages/flyouts/newPackage/newPackage.js]: Will add in future. I just put a TODO tag for now.

File: src/components/pages/packages/flyouts/newPackage/newPackage.scss [hunk 8]
```diff
@@ -0,0 +1 @@
+// Copyright (c) Microsoft. All rights reserved.
```
[the remark this answers is not in the document]
Comment (isaac73) [reply src/components/pages/packages/flyouts/newPackage/newPackage.scss]: Will add in future. I just put a TODO tag for now.

File: src/components/pages/packages/packages.js [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+ } from 'components/shared';
+ import { NewPackage } from './flyouts';
+ import { svgs } from 'utilities';
+
+import './packages.css';
+
+const closedFlyoutState = { openFlyoutName: undefined };
+
+export class Packages extends Component {
+
```
[the remark this answers is not in the document]
Comment (isaac73) [reply src/components/pages/packages/packages.js]: removed

File: src/components/pages/packages/packages.scss [hunk 10]
```diff
@@ -0,0 +1 @@
+// Copyright (c) Microsoft. All rights reserved.
```
[the remark this answers is not in the document]
Comment (isaac73) [reply src/components/pages/packages/packages.scss]: Will add in future. I just put a TODO tag for now.

File: src/components/pages/packages/packages.test.js [hunk 11]
```diff
@@ -0,0 +1 @@
+// Copyright (c) Microsoft. All rights reserved.
```
[the remark this answers is not in the document]
Comment (isaac73) [reply src/components/pages/packages/packages.test.js]: Will add in future. I just put a TODO tag for now.

File: src/components/pages/packages/packagesGrid/packagesGrid.scss [hunk 12]
```diff
@@ -0,0 +1,3 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+// TODO
```
[the remark this answers is not in the document]
Comment (isaac73) [reply src/components/pages/packages/packagesGrid/packagesGrid.scss]: Will add in future. I just put a TODO tag for now.

File: src/store/rootReducer.js [hunk 13]
```diff
@@ -8,12 +8,14 @@ import { reducer as appReducer } from './reducers/appReducer';
...
 import { reducer as devicesReducer } from './reducers/devicesReducer';
 import { reducer as rulesReducer } from './reducers/rulesReducer';
+import { reducer as packagesReducer } from './reducers/packagesReducer';
 
 const rootReducer = combineReducers({
   ...exampleReducer,
   ...appReducer,
   ...devicesReducer,
   ...rulesReducer,
+  ...packagesReducer,
```
[the remark this answers is not in the document]
Comment (isaac73) [reply src/store/rootReducer.js]: done

File: src/utilities/svgs.js [hunk 14]
```diff
@@ -60,6 +61,7 @@ export const svgs = {
     devices: TabDevicesIconPath,
     maintenance: TabMaintenanceIconPath,
     rules: TabRulesIconPath,
+    packages: PackagesIconPath,
```
[the remark this answers is not in the document]
Comment (isaac73) [reply src/utilities/svgs.js]: done

File: src/components/pages/packages/flyouts/index.js [hunk 15]
```diff
@@ -0,0 +1 @@
+export * from './newPackage';
```
Comment (mechaffin) [inline src/components/pages/packages/flyouts/index.js]: Copyright missing.

File: src/components/pages/packages/flyouts/newPackage/index.js [hunk 16]
```diff
@@ -0,0 +1 @@
+export * from './newPackage';
```
Comment (mechaffin) [inline src/components/pages/packages/flyouts/newPackage/index.js]: Copyright missing.

File: src/components/pages/packages/flyouts/newPackage/newPackage.js [hunk 17]
```diff
@@ -0,0 +1,13 @@
...
+import {  LinkedComponent} from 'utilities';
+
+import './newPackage.css';
+
+export class NewPackage extends LinkedComponent {
+// TODO
+
+render () {
+  return ([]);
+}
```
Comment (mechaffin) [inline src/components/pages/packages/flyouts/newPackage/newPackage.js]: nit: spacing... but yes, I know this whole file will change in your next checkin.

File: src/components/pages/packages/packages.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (mechaffin) [reply src/components/pages/packages/packages.js]: Then remove the ManageDeviceGroupsBtn

[the remark this answers is not in the document]
Comment (isaac73) [reply src/components/pages/packages/packages.js]: But Manage Device Group flout is required.

File: src/components/pages/packages/flyouts/index.js (hunk 15 above)
Comment (isaac73) [reply src/components/pages/packages/flyouts/index.js]: fixed

File: src/components/pages/packages/flyouts/newPackage/index.js (hunk 16 above)
Comment (isaac73) [reply src/components/pages/packages/flyouts/newPackage/index.js]: fixed

File: src/components/pages/packages/flyouts/newPackage/newPackage.js (hunk 17 above)
Comment (isaac73) [reply src/components/pages/packages/flyouts/newPackage/newPackage.js]: fixed

File: src/components/pages/packages/packages.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (mechaffin) [reply src/components/pages/packages/packages.js]: No, it is not. See the jobs detail page under Maintenance. If the device groups aren't relevant, then neither the drop down nor the manage button should be there.

[the remark this answers is not in the document]
Comment (isaac73) [reply src/components/pages/packages/packages.js]: Jobs and jobs details pages will be re-done shortly.

**Pull request merged**